core/trino-main/src/main/java/io/trino/connector/system/CatalogSystemTable.java

```diff
@@ -72,7 +75,10 @@ public RecordCursor cursor(ConnectorTransactionHandle transactionHandle, Connect
     {
         Session session = ((FullConnectorSession) connectorSession).getSession();
         Builder table = InMemoryRecordSet.builder(CATALOG_TABLE);
-        for (CatalogInfo catalogInfo : listCatalogs(session, metadata, accessControl)) {
+        List<CatalogInfo> catalogInfos = listCatalogs(session, metadata, accessControl).stream()
```

Comment:
Why not fix `listCatalogs` instead, what if other place use `listCatalogs` ?

Reply:
> I think that changing a listCatalogs semantics is pretty fragile. Filtering out on the call site is more explicit about intent


Exactly my way of thinking